main/src/ca/uwaterloo/flix/api/effectlock/HashType.scala

```diff
@@ -0,0 +1,146 @@
+    case TypeConstructor.Arrow(arity) => ???
+    case TypeConstructor.ArrowWithoutEffect(arity) => ???
+    case TypeConstructor.RecordRowEmpty => ???
+    case TypeConstructor.RecordRowExtend(label) => ???
+    case TypeConstructor.Record => ???
+    case TypeConstructor.Extensible => ???
+    case TypeConstructor.SchemaRowEmpty => ???
+    case TypeConstructor.SchemaRowExtend(pred) => ???
+    case TypeConstructor.Schema => ???
+    case TypeConstructor.Sender => ???
+    case TypeConstructor.Receiver => ???
+    case TypeConstructor.Lazy => ???
+    case TypeConstructor.Enum(sym, kind) => ???
+    case TypeConstructor.Struct(sym, kind) => ???
+    case TypeConstructor.RestrictableEnum(sym, kind) => ???
+    case TypeConstructor.Native(clazz) => ???
+    case TypeConstructor.JvmConstructor(constructor) => ???
+    case TypeConstructor.JvmMethod(method) => ???
+    case TypeConstructor.JvmField(field) => ???
+    case TypeConstructor.Array => ???
+    case TypeConstructor.ArrayWithoutRegion => ???
+    case TypeConstructor.Vector => ???
+    case TypeConstructor.Tuple(arity) => ???
+    case TypeConstructor.Relation(arity) => ???
+    case TypeConstructor.Lattice(arity) => ???
+    case TypeConstructor.True => ???
+    case TypeConstructor.False => ???
+    case TypeConstructor.Not => ???
+    case TypeConstructor.And => ???
+    case TypeConstructor.Or => ???
+    case TypeConstructor.Pure => ???
+    case TypeConstructor.Univ => ???
+    case TypeConstructor.Complement => ???
+    case TypeConstructor.Union => ???
+    case TypeConstructor.Intersection => ???
+    case TypeConstructor.Difference => ???
+    case TypeConstructor.SymmetricDiff => ???
+    case TypeConstructor.Effect(sym, kind) => ???
+    case TypeConstructor.CaseComplement(sym) => ???
+    case TypeConstructor.CaseUnion(sym) => ???
+    case TypeConstructor.CaseIntersection(sym) => ???
+    case TypeConstructor.CaseSymmetricDiff(sym) => ???
+    case TypeConstructor.CaseSet(syms, enumSym) => ???
+    case TypeConstructor.Region(sym) => ???
+    case TypeConstructor.RegionToStar => ???
+    case TypeConstructor.RegionWithoutRegion => ???
+    case TypeConstructor.Error(id, kind) => throw InternalCompilerException("Unexpected error type constructor", SourceLocation.Unknown)
+  }
+
+  private def hashKindedTypeVarSym(sym0: Symbol.KindedTypeVarSym): Array[Byte] = ???
+
+  private def hashBytes(bytes: Array[Byte]): Array[Byte] = {
+    val md = MessageDigest.getInstance("SHA-256")
+    md.update(bytes)
+    md.digest()
+  }
+
+  private def hashByte(byte: Byte): Array[Byte] = {
+    val md = MessageDigest.getInstance("SHA-256")
+    md.update(byte)
+    md.digest()
+  }
+
+  /**
+    * Common super type for hashable types.
+    *
+    * Case objects have hash codes that can be converted to byte values
+    * and thus represents a stable and unique byte value for the constructor.
+    */
+  private sealed trait HType
```

Comment:
Why do we need this? Can we not just perform a recursive traversal over the datatype?